flume/src/main/java/es/tid/fiware/fiwareconnectors/cygnus/sinks/OrionSink.java

```diff
@@ -137,22 +137,35 @@ public Status process() throws EventDeliveryException {
             // rollback only if the exception is about a persistence error
             if (e instanceof CygnusPersistenceError) {
                 logger.error(e.getMessage());
-                
+
                 // check the event HEADER_TTL
-                int ttl = new Integer(event.getHeaders().get(Constants.HEADER_TTL)).intValue();
+                int ttl;
+                String ttlStr = event.getHeaders().get(Constants.HEADER_TTL);
                 
-                if (ttl > 0) {
-                    String newTTL = new Integer(ttl - 1).toString();
-                    event.getHeaders().put(Constants.HEADER_TTL, newTTL);
+                try {
+                    ttl = Integer.parseInt(ttlStr);
+                } catch (NumberFormatException nfe) {
+                    ttl = 0;
+                    logger.error("Invalid TTL value (id=" + event.hashCode() + ", ttl=" + ttlStr
+                          +  ", " + nfe.getMessage() + ")");
+                } // try catch
+                
+                if (ttl == -1) {
```

Comment:
Fixed in https://github.com/telefonicaid/fiware-connectors/commit/4f731fe50691d4b39c4d173d6a650c37fe72f7ed